domain/agentprovisioner/service/service.go

```diff
@@ -0,0 +1,272 @@
+	}
+}
+
+// ContainerManagerConfigForType returns the container manager config for the
+// given container type.
+func (s *Service) ContainerManagerConfigForType(
+	ctx context.Context,
+	containerType instance.ContainerType,
+) (containermanager.Config, error) {
+	rval := containermanager.Config{}
+
+	modelID, err := s.st.ModelID(ctx)
+	if err != nil {
+		return rval, fmt.Errorf("getting ID for current model: %w", err)
+	}
+
+	cfg, err := s.st.GetModelConfigKeyValues(ctx,
+		config.LXDSnapChannel,
+		config.ContainerImageMetadataURLKey,
+		config.ContainerImageMetadataDefaultsDisabledKey,
+		config.ContainerImageStreamKey,
+	)
+	if err != nil {
+		return rval, fmt.Errorf(
+			"cannot get model config keys when calculating container manager config: %w",
+			err,
+		)
+	}
+
+	rval.ModelID = modelID
+	if containerType == instance.LXD {
+		rval.LXDSnapChannel = cfg[config.LXDSnapChannel]
+	}
+	rval.ImageMetadataURL = cfg[config.ContainerImageMetadataURLKey]
+	rval.MetadataDefaultsDisabled, _ = strconv.ParseBool(cfg[config.ContainerImageMetadataDefaultsDisabledKey])
+	rval.ImageStream = cfg[config.ContainerImageStreamKey]
+
+	return rval, nil
+}
+
+// ContainerNetworkingMethod determines the container networking method that
+// should be used, based on the model config key "container-networking-method"
+// and the current provider.
+func (s *Service) ContainerNetworkingMethod(ctx context.Context) (containermanager.NetworkingMethod, error) {
+	cfg, err := s.st.GetModelConfigKeyValues(ctx, config.ContainerNetworkingMethod)
+	if err != nil {
+		return "", fmt.Errorf("getting container networking method from model config: %w", err)
+	}
+
+	method := containermanager.NetworkingMethod(cfg[config.ContainerNetworkingMethod])
+	switch method {
+	case containermanager.NetworkingMethodLocal, containermanager.NetworkingMethodProvider:
+		return method, nil
+	case "":
+		// Auto-configure container networking method below
+	default:
+		return "", fmt.Errorf("invalid container networking method %q in model config", method)
+	}
+
+	provider, err := s.providerGetter(ctx)
+	if errors.Is(err, errors.NotSupported) {
+		// Provider doesn't have the SupportsContainerAddresses method
+		return containermanager.NetworkingMethodLocal, nil
+	}
+	if err != nil {
+		return "", fmt.Errorf(
+			"cannot get networking provider for model: %w",
+			err,
+		)
+	}
+
+	supports, err := provider.SupportsContainerAddresses(envcontext.WithoutCredentialInvalidator(ctx))
+	if errors.Is(err, errors.NotSupported) {
+		return containermanager.NetworkingMethodLocal, nil
+	}
+	if err != nil {
+		return "", fmt.Errorf(
+			"cannot determine if provider supports container addresses: %w",
+			err,
+		)
+	}
+	if supports {
+		return containermanager.NetworkingMethodProvider, nil
+	}
+	return containermanager.NetworkingMethodLocal, nil
+}
+
+// ContainerConfig returns the container config for the model.
+func (s *Service) ContainerConfig(ctx context.Context) (params.ContainerConfig, error) {
```

Comment:
Yes this should be a new type in core.

Reply:
What's the reason for this? Because I would literally be copy-pasting the type from params to another package and then copying all the struct fields in the facade method